include/fc/static_variant.hpp

```diff
@@ -308,32 +338,54 @@ class static_variant {
     }
     template<typename visitor>
     typename visitor::result_type visit(visitor& v) {
-        return impl::storage_ops<0, Types...>::apply(_tag, storage, v);
+        return visit( _tag, v, (void*) storage );
     }
 
     template<typename visitor>
     typename visitor::result_type visit(const visitor& v) {
-        return impl::storage_ops<0, Types...>::apply(_tag, storage, v);
+        return visit( _tag, v, (void*) storage );
     }
 
     template<typename visitor>
     typename visitor::result_type visit(visitor& v)const {
-        return impl::storage_ops<0, Types...>::apply(_tag, storage, v);
+        return visit( _tag, v, (const void*) storage );
     }
 
     template<typename visitor>
     typename visitor::result_type visit(const visitor& v)const {
-        return impl::storage_ops<0, Types...>::apply(_tag, storage, v);
+        return visit( _tag, v, (const void*) storage );
     }
 
-    template<tag_type w, typename visitor>
-    static typename visitor::result_type visit(visitor& v) {
-        return impl::storage_ops<0, Types...>::apply((const tag_type)w, (const void*)nullptr, v);
+    template<typename visitor>
+    static typename visitor::result_type visit( tag_type tag, visitor& v, void* data )
+    {
+        static std::vector<std::function<typename visitor::result_type(visitor&,void*)>> wrappers = init_wrappers<visitor,void*,Types...>();
+        FC_ASSERT( tag < count(), "Unsupported type ${tag}!", ("tag",tag) );
```

Comment:
Assert `tag >= 0` as well?

BTW please wrap long lines (perhaps can use `auto` here)? (BTW just found that Github in Firefox in Ubuntu desktop 18.04 only shows 123 characters a line, while in Chrome in Windows 7 it shows 130 characters)